### ISSUE TITLE
iquest parser: this should error, but returned nil

### ISSUE THREAD

**Issue by mcast** [irods/irods#5734]
- [x] main
- [x] 4-3-stable

Using icommands and server 4.2.7 on Bionic.

I fed junk to `iquest`, I think it should have failed, but it gave me nothing.

It's not a problem for me, so I'm writing the issue in case there are other problems hiding nearby and maybe this will bring them out.

----

While debugging I repurposed a generated command, botched the edit, and asked for

```
(cgpdev)mca@casm3-head2:~$ iquest  --no-page  -z  /cgp_dev  "%s^A      %s^A    %s^A    %s^A    %s^A    %s^A    %s^B^B"  "select DATA_ID, DATA_REPL_NUM, COLL_NAME, DATA_NAME, DATA_RESC_HIER, DATA_PATH, META_DATA_ATTR_VALUE where META_DATA_ATTR_NAME = 'id_foo and META_DATA_ATTR_VALUE in ('260,'3261852','3261856','3261901','3080907','3083125','3083853','3084203','3085046','3091021','3092210','3092313','3093766','3094073','3094078','3095017','3095445','3095522','3097128','3097225','3097311','3097480','3097702','3097750')"
Zone is /cgp_dev
CAT_NO_ROWS_FOUND: Nothing was found matching your query
(cgpdev)mca@casm3-head2:~$ echo $?
0
```

Well that's odd because I just put a file with an id_foo AVU. But there are two missing `'`s, resulting in this query

```
select [...] where META_DATA_ATTR_NAME = 'id_foo and META_DATA_ATTR_VALUE in ('
260,'3261852','3261856','3261901','3080907','3083125','3083853','3084203','3085046','3091021','3092210','3092313','3093766','3094073','3094078','3095017','3095445','3095522','3097128','3097225','3097311','3097480','3097702','3097750')"
```

The second line appears to be syntactically complete junk and I can't guess why it didn't provoke an error. Of course there are no AVUs with ATTR_NAME `id_foo and META_DATA_ATTR_VALUE in (` so if it does run there won't be any rows back...  but how did it get to run?

----

Thanks for reading,  
m

**Comment by mcast** [comment]
Thanks @trel , good to hear it's on the radar and the `spLogSql=1` tip may be useful later too.

This points to a simpler bug in the same line,

```
mca@casm3-head2:~$ iquest '%s' "select DATA_NAME where DATA_ID = '1234')"
CAT_NO_ROWS_FOUND: Nothing was found matching your query
mca@casm3-head2:~$ iquest '%s' "select DATA_NAME where DATA_ID = '1234'"
CAT_NO_ROWS_FOUND: Nothing was found matching your query
```

**Comment by trel** [comment]
With a valid DATA_ID... I would have expected a parser error on the first.

```
$ iquest '%s' "select DATA_NAME where DATA_ID = '10502')"                                                         
foo0048
$ iquest '%s' "select DATA_NAME where DATA_ID = '10502'"                                                          
foo0048
```

With an invalid DATA_ID... I also would have expected a parser error on the first.
```
$ iquest '%s' "select DATA_NAME where DATA_ID = '999')"                                                           
CAT_NO_ROWS_FOUND: Nothing was found matching your query
$ iquest '%s' "select DATA_NAME where DATA_ID = '999'"                                                            
CAT_NO_ROWS_FOUND: Nothing was found matching your query
```

**Comment by korydraughn** [comment]
Resolved by the GenQuery2 parser (#7570).

Just needs tests.
